### ISSUE TITLE
[Accessibility]: Code Insights Creation UI - Data series "Done" button too verbose

### ISSUE THREAD

**Issue by unclejustin** [sourcegraph/sourcegraph#41668]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33494

### Problem description

The current aria-label for the "Done" button on a data series is "Submit button for data series," which causes the screen reader to say "Submit button for data series button." which is redundant and incorrect. This is not a submit button.

### Expected behavior

Recommend changing the aria-label to "Data series done editing" which the screen ready reads "Data series done editing button"

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by felixfbecker** [comment]
The label should generally only contain the action, if a verb in imperative form. It shouldn't rely on the screen reader announcing the role after the label, because not all screen readers may do that.
"Submit button for data series" is a description, but not a label in imperative form.
"Data series done editing" also doesn't fit this form.

Something like "Confirm data series" could work. But the whole data series sub-form should probably also be in some sort of `group` so that the nested hierarchy is communicated, so it's somewhat obvious that the button applies to the data series, as opposed to the label having to redundantly state it.

**Comment by felixfbecker** [comment]
Bumping this up to a violation – this issue is actually a violation of [2.5.3: Label in Name](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html). The text of the button needs to be equal or a substring of the `aria-label`, they are currently fully distinct.
